shell/platform/android/platform_view_android.cc

```diff
@@ -107,7 +107,9 @@ PlatformViewAndroid::PlatformViewAndroid(
       jni_facade_(jni_facade),
       platform_view_android_delegate_(jni_facade) {}
 
-PlatformViewAndroid::~PlatformViewAndroid() = default;
+PlatformViewAndroid::~PlatformViewAndroid() {
+  surface_factory_->SetExternalViewEmbedder(nullptr);
```

Comment:
surface_factory_ is a shared_ptr, you could be editing a variable that is being held by someone unexpectedly.  I don't see anywhere where the surface_factory_ is actually shared.  If it isn't shared we should move it to a unique_ptr to make this operation obviously safe.

Reply:
Ahh wait, the AndroidExternalViewEmbedder is holding onto it.  Do you think we should make that a weak_ptr instead?

Reply:
Moved the strong reference to the ExternalViewEmebedder up to the PlatformView.
Changed the SurfaceFactory's reference to a weak pointer.